receiver/elasticsearchreceiver/factory.go

```diff
@@ -35,12 +35,18 @@ func createDefaultConfig() component.Config {
 	cfg := scraperhelper.NewDefaultControllerConfig()
 	cfg.CollectionInterval = defaultCollectionInterval
 
+	clientConfig := confighttp.NewDefaultClientConfig()
+	clientConfig.Endpoint = defaultEndpoint
+	clientConfig.Timeout = defaultHTTPClientTimeout
+	clientConfig.MaxIdleConns = nil
+	clientConfig.MaxIdleConnsPerHost = nil
+	clientConfig.MaxConnsPerHost = nil
+	clientConfig.IdleConnTimeout = nil
+	clientConfig.Headers = nil
+
```

Comment:
```
	clientConfig.MaxIdleConns = nil
	clientConfig.MaxIdleConnsPerHost = nil
	clientConfig.MaxConnsPerHost = nil
	clientConfig.IdleConnTimeout = nil
	clientConfig.Headers = nil
```

This part can be deleted, for MaxIdleConns, MaxIdleConnsPerHost, MaxConnsPerHost, IdleConnTimeout even if you set to nil here, the same defaults end up being used in the call to ToClient: https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L136.

For headers, ToClient checks for len (https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L210), so whether it's nil or an empty map (set in NewDefaultClientConfig) behaviour should be the same.

Reply:
Good point, thanks for the detailed explanation. Removed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35509/commits/a9396e056d676e6ed2f746d9dc3d7ee50edd1b54